protocol/xmrtaker/message_handler.go

```diff
@@ -296,32 +296,10 @@ func (s *swapState) handleNotifyXMRLock(msg *message.NotifyXMRLock) (net.Message
 		select {
 		case <-s.ctx.Done():
 			return
+		// TODO: document why we add one second
 		case <-time.After(until + time.Second):
-			s.Lock()
-			defer s.Unlock()
-
-			if !s.info.Status().IsOngoing() {
-				return
-			}
-
-			// XMRMaker hasn't claimed, and we're after t_1. let's call Refund
-			txhash, err := s.refund()
-			if err != nil {
-				log.Errorf("failed to refund: err=%s", err)
-				return
-			}
-
-			log.Infof("got our ETH back: tx hash=%s", txhash)
-			s.clearNextExpectedMessage(types.CompletedRefund) // TODO: duplicate?
-
-			// send NotifyRefund msg
-			if err = s.SendSwapMessage(&message.NotifyRefund{
-				TxHash: txhash.String(),
-			}, s.ID()); err != nil {
-				log.Errorf("failed to send refund message: err=%s", err)
-			}
-
-			_ = s.Exit()
```

Comment:
This was the problem line grabbing the lock a second time.  I separated out an `exit` function (does not grab lock) from `Exit` (grabs lock) to match the existing implementation in `xmrmaker`. The code above was moved into `handleT1Expired` and that function now calls `s.exit` instead of `s.Exit`.